File: 0169. Majority Element.cpp
//Time Complexity: O(nlogn)
//Space Complexity: O(n)
//Approach: Sorting
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        return nums[nums.size() / 2];  
    }
};

File: 0169. Majority Element.cs
// Time Complexity: O(n)
// Space Complexity: O(n)
// Approach: Hashmap
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int, int> freq = new Dictionary<int, int>();
        int majorityFreq = nums.Count() / 2;
        for(int i = 0; i < nums.Count(); i++){
            int num = nums[i];
            if(freq.ContainsKey(num)){
                freq[num]++;
            } else{
                freq[num] = 1;
            }
            if(freq[num] > majorityFreq){
                return num;
            }
        }
        return 0;
    }
}
